### ISSUE TITLE
The ember-source version

### ISSUE THREAD

**Issue by rreckonerr** [sly7-7/bug-query-params#1]
According to the `package.json` you've used the latest release of `ember-source`: "3.26.0-beta.2"`, but it's different from the master branch. While I was able to reproduce that the bug is gone on ember 3.27 (which is not released yet).

So there's no other way to reproduce it other than symlinking the ember.js repo I believe. Correct me if I'm wrong :)

**Comment by sly7-7** [comment]
@rreckonerr 

Yeah, locally, I've `yarn linked` against ember's master branch, but this does not reflect in the package.json, neither in the yarn.lock file.

To be precise, what I've just done is:

ember directory:
  git checkout v3.27.0-alpha.1 (which put the repo in a `detach head` state on 
![Capture d’écran 2021-03-02 à 15 07 34](https://user-images.githubusercontent.com/1826661/109660343-06280280-7b69-11eb-87c8-e3cae51bf014.png)

rm -rf dist node_modules
yarn
yarn run build
yarn link

then in my app directory:

yarn link ember-source
yarn 
ember s

![Capture d’écran 2021-03-02 à 15 09 31](https://user-images.githubusercontent.com/1826661/109660546-4be4cb00-7b69-11eb-9a3e-a21329179677.png)

![Capture d’écran 2021-03-02 à 15 10 09](https://user-images.githubusercontent.com/1826661/109660646-66b73f80-7b69-11eb-84d5-dfb796430a10.png)

Here is the "result": 

https://user-images.githubusercontent.com/1826661/109660823-95351a80-7b69-11eb-9ef5-0c4bab775bfa.mov

First I click to set the query param to false. You can see the list is in reverse order.
Then when I click on an item, during the loading phase, the list is reordered (which mean the qp is reset to its default value), and then, after the transition completes, it comes back to its expected state.

I hope, it's understandable.
Also, I could mention, that if I remove the loading template, this 'intermediate' incorrect state disappear

**Comment by rreckonerr** [comment]
Hah, alright, let's make sure that we're on the same ember version here.

If you were following my instructions, then in your Ember Inspector's Info tab you should see `3.25.0-canary+1fa40fb5`, where `1fa40fb5` stands for https://github.com/emberjs/ember.js/commit/1fa40fb5ed3080f8987e8691136b070a00664de5

I just cloned your repo and can confirm that the bug is gone on `3.25.0-canary+1fa40fb5` AKA `3.27.0-alpha.1` branch. But there's still possibility that I've screwed up something symlinking the stuff :)

> then in my app directory:
yarn link ember-source
yarn
ember s

I'm not 100% sure that it'll help, but can you also `yarn install --force` after ☝️ in `bug-query-params` repo and check your ember inspector if you're on `3.25.0-canary+1fa40fb5`?

Also, `la ~/.config/yarn/link` will show you all linked packages.

**Comment by sly7-7** [comment]
🤔  In your ember.js repo could you `rm -rf dist` and re-run ember build ? Because I clearly not understand why 3.27.alpha.1 is not showing into Ember inspector tab.
That's a git log, and the tag 3.27.alpha1 is very recent

![Capture d’écran 2021-03-02 à 16 03 15](https://user-images.githubusercontent.com/1826661/109667813-ce24bd80-7b70-11eb-92a4-db18a8396368.png)

comparing to the commit you've mentioned.

![Capture d’écran 2021-03-02 à 16 04 58](https://user-images.githubusercontent.com/1826661/109668042-0b894b00-7b71-11eb-9e1d-5cc6c830690d.png)

I also may be wrong, but I think your ember version, is before the bug has been introduced.

**Comment by rreckonerr** [comment]
> I also may be wrong, but I think your ember version, is before the bug has been introduced.

Hmm, exactly what I was afraid of. I will try to make sure I'm on the right version of ember. 

Btw, what version of ember do you get in ember inspector?

**Comment by sly7-7** [comment]
Yup, because on a fresh ember.js clone, if I don't `ember build` ember.js, I have

![Capture d’écran 2021-03-02 à 16 13 50](https://user-images.githubusercontent.com/1826661/109669448-62dbeb00-7b72-11eb-8e23-7f3b7360164d.png)

**Comment by sly7-7** [comment]
https://user-images.githubusercontent.com/1826661/109660646-66b73f80-7b69-11eb-84d5-dfb796430a10.png

**Comment by sly7-7** [comment]
> Btw, what version of ember do you get in ember inspector?
https://user-images.githubusercontent.com/1826661/109660646-66b73f80-7b69-11eb-84d5-dfb796430a10.png

**Comment by sly7-7** [comment]
> Btw, what version of ember do you get in ember inspector?


![Capture d’écran 2021-03-02 à 16 18 09](https://user-images.githubusercontent.com/1826661/109669959-e8f83180-7b72-11eb-8691-51b7acee6b70.png)

**Comment by rreckonerr** [comment]
Yup! Just removed ember.js folder and cloned it again. Wonder if my recent move to a new laptop screwed it up.

So unfortunately the bug is still there. Thanks for helping to figure things out!

**Comment by sly7-7** [comment]
No problem :) I've pinged @rwjblue and @chancancode on the discord channel about the issue.

Also, I think the bug was not catched because I think it's hard to test this "intermediate" state during loading the model.

**Comment by rreckonerr** [comment]
I hope someone will take a look at it! Query param bugs were there for ages. I got rid of some at one point, but this unfortunately introduced this issue you are facing now.
